### PR TITLE
chore: Update Go to 1.18.1

### PR DESCRIPTION
**Pull request by sspaink**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -230,10 +230,10 @@ plugin-%:
 	@echo "Starting dev environment for $${$(@)} input plugin..."
 	@docker-compose -f plugins/inputs/$${$(@)}/dev/docker-compose.yml up
 
-.PHONY: ci-1.17
-ci-1.17:
-	docker build -t quay.io/influxdb/telegraf-ci:1.17.7 - < scripts/ci-1.17.docker
-	docker push quay.io/influxdb/telegraf-ci:1.17.7
+.PHONY: ci-1.18
```
Comment (powersj) [inline Makefile:233]: if we are only maintaining one version, any reason to note rename this to something version independent? e.g. `ci`

File: Makefile [hunk 2]
```diff
@@ -230,10 +230,10 @@ plugin-%:
...
+ci-1.18:
+	docker build -t quay.io/influxdb/telegraf-ci:1.18.0 - < scripts/ci-1.18.docker
```
Comment (powersj) [inline Makefile:235]: same question with the `ci.docker` file

**Review by powersj — COMMENTED** [review]
I've been hesitant to do this, but it sounds like there haven't been any major issues for the first couple weeks of 1.18, so probably safe.

Two questions about naming of file and a make target and then good to go.

File: Makefile (hunk 1 above)
Comment (sspaink) [reply Makefile]: Good point! Sounds like a good change, two less things to update when going to a new version. I've pushed a change.

**Review by powersj — APPROVED** [review]
Thank you!

**Comment by sspaink** [comment]
Looks like the Darwin version hasn't been released yet for v1.18.1: https://go.dev/dl/
As soon as that is released, I can update the SHA's in the script.

**Review by powersj — APPROVED** [review]
Thanks for making the update to v1.18.1!

**Pull request merged**